### DISCUSSION TITLE
ref.refresh example

### DISCUSSION THREAD

**Discussion by felipe-xavier** [rrousselGit/riverpod#2060] [Q&A] [answered]
Please, could someone provide an example of ref.refresh where the needed return value is used? I couldn't find a use case, even inside the `RefreshIndicator` widget, the `onRefresh` field expects a `Future<void> Function()`.

Thanks in advance.

**Comment by mafreud** [comment]
I have the same question. What do you think?
@AhmedLSayed9

**Reply by mafreud** [reply]
You can do this.
```dart
onRefresh: () => ref.refresh(futureProvider.future),
```

**Comment by AhmedLSayed9** [comment, marked as answer]
It's just a shorthand for invalidating a provider then return the created value by using `ref.read`. Its common use-case by using `ref.refresh(provider.future)` with `RefreshIndicator`.

as `provider.future` obtains a Future that resolves with the first state value that is not AsyncLoading, so it's like you set state to loading (i.e by using ref.invalidate) and then return a future that resolves when data/error state is emitted:
```dart
onRefresh: (){
  ref.invalidate(futureProvider); //provider is invalidated so state is back to loading
  return ref.read(futureProvider.future); //return future that resolves when data/error state is emitted so the refresh indicator keeps loading until then
},
```
That's identical to:
```dart
onRefresh: () => ref.refresh(futureProvider.future),
```

**Reply by mafreud** [reply]
Thank you!
